src/layerstack.stanza

```diff
@@ -292,6 +316,49 @@ public defn LayerStack (
   ) -> LayerStack:
   #LayerStack(name, description, to-vector<LayerSpec>(initial-layers))
 
+defmethod write (o:OutputStream, s:LayerStack) :
+  println(o, "LayerStack: %_" % [value?(name(s))])
+  println(o, "  Layers: %_ copper layers %_ total layers" % [get-conductor-count(s) length(layers(s))])
+  for [idx, l] in enumerate(layers(s)) do:
+    println(o, "    %_ : %_ | %_ | %_" %
+      [idx, value?(name(l)) get-type(material(l)) thickness(l)])
+
+doc:\<DOC>
+  Usage: for [idx, line] in enumerate(lines) do: 
+           println(“Line %: %_” % [idx, line])
+<DOC>
+defn enumerate<?T> (s:Seqable<?T>) -> Seq<[Int, T]> :
+  var i = 0
+  for x in s seq :
+    next where:
+      val next = [i, x]
+      i = i + 1
+
+doc:\<DOC>
+  Get a layer by index
+  Usage: stack[idx]
+<DOC>
+public defn get (stack:LayerStack, idx:Int) -> LayerSpec :
+  layers(stack)[idx]
+
+doc:\<DOC>
+  Modify a layer by index
+  Usage: stack[idx] - new-layer
+<DOC>
+public defn set (stack:LayerStack, idx:Int, layer:LayerSpec) -> False :
+  layers(stack)[idx] = layer
+  ;<A>
+  set-layers(stack new-layers) where :
+    val new-layers = to-tuple $ cat-all $
+      [layers(stack)[0 to idx] [layer] layers(stack)[(idx + 1) to false]]
+  ;<A>
+
+doc:\<DOC>
+  Get the idx-th conductor layer
+  Usage: conductors(stack)[idx]
+<DOC>
+public defn conductors (stack:LayerStack) -> Tuple<LayerSpec> :
+  to-tuple $ filter({material(_) is ConductorMaterial} layers(stack))
```

Comment:
This is good :+1:

Reply:
Come to think of it: `layers(stack)` returns a `Vector`.
But `conductors(stack)` would return a tuple.
This is not consistent. But maybe it does not matter much.

`layers` in `LayerStack` only needs to be a vector internally.

Reply:
I think immutability of the `Tuple` here is a good thing.